lib/tasks/journeys_report.rake

```diff
@@ -2,60 +2,97 @@ require 'csv'
+    last_run_date = DateTime.strptime('0', '%s')
+  end
+  last_run_date
+end
+
+def update_bookmark(bucket_name)
+  s3 = Aws::S3::Resource.new
+  bookmark = { 'last_run': DateTime.now.utc.to_s }
+  s3.bucket(bucket_name).put_object(key: 'bookmark', body: JSON[bookmark])
+end
+
 namespace :journeys do
   desc 'Exports a JSON report of all journeys'
   task json_report: :environment do
     timestamp = DateTime.now.utc.to_i
+    bucket_name = ENV['S3_REPORTING_BUCKET_NAME'] || raise('S3_REPORTING_BUCKET_NAME not defined')
+
+    if ENV['RAILS_ENV'].to_s == 'development' || ENV['RAILS_ENV'].to_s == ''
+      Aws.config.update(
+        endpoint: 'http://localhost:4572',
+        credentials: Aws::Credentials.new('fakeid', 'fakesecret'),
+        region: 'eu-west-2',
+        force_path_style: true,
+      )
+    end
+
+    last_report_date = get_last_run(bucket_name)
+    puts last_report_date
 
     Supplier.order(:key).each do |supplier|
       filename = "#{supplier.key}-#{timestamp}.json"
       puts "Generating #{filename}..."
       moves = []
       Move
           .where("ID in (select move_id from journeys WHERE supplier_id = '#{supplier.id}')")
+          .where("moves.date > '#{last_report_date}'")
+          .where('moves.date < now()')
           .order('moves.date ASC, moves.reference ASC').each do |move|
         journeys = []
         Journey.where(move: move).default_order.each do |journey|
           journeys << {
-            timestamp: journey.client_timestamp.strftime('%d/%m %H:%M:%S'),
+            supplier: supplier.name,
+            timestamp: journey.client_timestamp.iso8601,
             from: {
-              title: journey.from_location.title,
-              id: journey.from_location.id,
+              id: journey.from_location.nomis_agency_id,
               nomis_agency_id: journey.from_location.nomis_agency_id,
             },
             to: {
-              title: journey.to_location.title,
-              id: journey.to_location.id,
+              id: journey.to_location.nomis_agency_id,
               nomis_agency_id: journey.to_location.nomis_agency_id,
             },
             vehicle: journey.vehicle['registration'],
             billable: journey.billable,
             events: journey.events.default_order.map do |event|
               {
-                timestamp: event.client_timestamp.strftime('%d/%m %H:%M:%S'),
+                timestamp: event.client_timestamp.iso8601,
                 event: event.event_name,
                 notes: event.notes,
-                from_location: event.from_location&.title,
-                to_location: event.to_location&.title,
+                from_location: event.from_location&.nomis_agency_id,
+                to_location: event.to_location&.nomis_agency_id,
               }
             end,
           }
         end
         moves << {
           supplier: supplier.key,
           reference: move.reference,
+          request_date: move.created_at.to_date,
```

Comment:
add a comment/note/todo that this should be fixed at some point; otherwise we will forget and one day someone will ask why the requested date is the same as the created date

Reply:
actually, I really don't think we should label this as `request_date` to avoid misunderstanding of data consumers.